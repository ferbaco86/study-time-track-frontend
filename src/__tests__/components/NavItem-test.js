import { render, screen, act } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../reducers/store';
import NavItem from '../../components/NavItem';

describe('Rendering component', () => {
  it('creates a NavItem component', () => {
    act(() => {
      render(
        <BrowserRouter>
          <Provider store={store}>
            <NavItem
              icon="fas fa-plus"
              legend="test"
              path="/test/path"
            />
          </Provider>
        </BrowserRouter>,
      );
    });
    screen.getByText('test');
  });
});

describe('NavItem', () => {
  it('renders correctly', () => {
    const snap = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <NavItem
            icon="fas fa-plus"
            legend="test"
            path="/test/path"
          />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
