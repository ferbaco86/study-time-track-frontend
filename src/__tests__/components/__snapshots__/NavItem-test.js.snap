// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavItem renders correctly 1`] = `
<a
  className="sc-bdfBwQ hoYbZe"
  href="/test/path"
  onClick={[Function]}
>
  <i
    className="sc-gsTCUz cknVKz fas fa-plus"
  />
  <small
    className="sc-dlfnbm eIUnSS"
  >
    test
  </small>
</a>
`;
